src/base/digest32.h

```diff
@@ -41,13 +43,15 @@ class Digest32
     using UnderlyingType = lt::digest32<N>;
 
     Digest32() = default;
+    Digest32(const Digest32 &other) = default;
```

Comment:
Is this implicitly shared only needed in v4.4.x? also it is not clear why you need it now.

Reply:
> it is not clear why you need it now.

Although `TorrentID` doesn't contain much data, it still takes up much more than a single pointer. We have to have multiple copies of the same ID, so it's a good idea to have `TorrentID` implicitly shared. In addition, a copy of all registered IDs is created in this PR in QSet to speed up the lookup.


> Is this implicitly shared only needed in v4.4.x?

No.
The same logic will be implemented in v4.5.

Reply:
>No.
The same logic will be implemented in v4.5.

And in master branch?

Reply:
> > No.
> > The same logic will be implemented in v4.5.
> 
> And in master branch?

Yes.
Until we decide that hardly anyone is using qBittorrent v4.4 (and below) and we no longer need to worry about supporting hybrid torrents stored with legacy filenames.